torch/nn/parallel/distributed.py

```diff
@@ -1352,6 +1381,24 @@ def _check_comm_hook(self, hook):
     def _distributed_rank(self):
         return dist.get_rank(self.process_group)
 
+    def _create_gloo_pg(self):
+        """
+        Creates a gloo process group with same ranks as self.process_group.
+        """
+        if _GLOO_AVAILABLE:
+            ranks = list(range(dist.get_world_size(self.process_group)))
+            cpu_pg = dist.new_group(
```

Comment:
actually I just realized that it is not good to call new_group() in DDP code base directly right now. as new_group() assumes users have created default process group using init_process_group(). but users does not have to call init_process_group() to use DDP, they could create a processGroup by directly using processGroupNCCL()...

so in case users are using processGroupNCCL() to create processGroup and pass it to DDP, calling new_group() will throw errors here.